### PR TITLE
Update msgpack-lite definitions

### PR DESCRIPTION
**Pull request by efokschaner**
@endel we'll probably need to discuss my changes to the Codec interface. I've removed anything that wasn't documented. If those changes break anything for you we should check with @kawanet whether he intended to expose and support members other than `addExtPacker` and `addExtUnpacker`. If that doesn't turn out to be the case then you can still augment the Codec interface in your local source.

DefinitelyTyped Checklist:
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/kawanet/msgpack-lite
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: msgpack-lite/index.d.ts [hunk 1]
```diff
@@ -1,14 +1,72 @@
-// Type definitions for msgpack-lite 0.1.20
+// Type definitions for msgpack-lite 0.1
```
Comment (efokschaner) [inline msgpack-lite/index.d.ts]: The linter insisted on no patch version.

**Comment by efokschaner** [comment]
Wil fix up the ci error later today.

**Comment by kawanet** [comment]
Thank you for mentioning me. I am pure JavaScript love and have less knowledge about TypeScript. I appreciate any helps in advance. 

The Codec class seems to have 5 methods:

```
- init (): void;
- addExtPacker (etype: number, Class: any, packer: (value: any) => any): void;
- getExtPacker (value: any): (value: any) => any;
- addExtUnpacker (etype: number, unpacker: (value: any) => any): void;
- getExtUnpacker (etype: number): (value: any) => any;
```

It is correct that `addExtPacker` and `addExtUnpacker` methods are roughly documented on README. The others however are not yet because they seem rarely needed at almost any cases in my guess. It'd be documented when needed.

---

Memorandum:

The Codec class may look bit strange and have complex implementation. 
It's designed as something like multiple mixins:

write-core.js installs addExtPacker, getExtPacker and init methods on it.
read-core.js installs addExtUnpacke, getExtUnpacker and init.

Some use-cases requires the encoder feature only. Some requires the decoder feature only, vice versa. Less number of dependency files helps to generate shorter code with browserify. I have a future plan to ship a pair of separated drops: msgpack-lite.encoder.js and msgpack-lite.decoder.js.

**Comment by efokschaner** [comment]
Thanks for your attention @kawanet. The details of the implementation are no problem at all. The only questions we have are about the interface for consumers of msgpack-lite.

- I don't think we need Codec to be directly constructible as there is `msgpack.createCodec` (so no `new` member needed in TS).
- Are the methods `init`, `getExtPacker` and `getExtUnpacker`, and the member `options`, intended for use or should they be considered "implementation detail" of msgpack-lite.

Let's remember that it's easy to expose things later if we decide to support them, and harder to do the opposite and hide interfaces we already exposed.

**Comment by efokschaner** [comment]
@kawanet just wanted to ping you regarding my 2 questions above.

**Comment by bowdenk7** [comment]
friendly ping @kawanet - would love to get this merged!

**Comment by endel** [comment]
IMHO this can be merged. I wrote the first version in a hurry, just to have the definitions. Now it looks much better. Great job @efokschaner 🎉

**Comment by kawanet** [comment]
> I don't think we need Codec to be directly constructible as there is msgpack.createCodec (so no new member needed in TS)?

Right. createCodec is enough.

> Are the methods init, getExtPacker and getExtUnpacker, and the member options, intended for use or should they be considered "implementation detail" of msgpack-lite?

They are, currently at least, private methods because I don't have any use cases to call those methods from applications.

This might be too late to comment here, however. Thanks anyway!

**Pull request merged**